ast/TreeEquality.cc

```diff
@@ -275,7 +275,16 @@ bool structurallyEqual(const void *avoid, const Tag tag, const void *tree, const
         case Tag::ConstantLit: {
             auto *a = reinterpret_cast<const ConstantLit *>(tree);
             auto *b = reinterpret_cast<const ConstantLit *>(other);
-            return a->symbol == b->symbol && structurallyEqual(avoid, a->original, b->original);
+            if (a->symbol != b->symbol) {
+                return false;
+            }
+            if (a->original && b->original) {
```

Comment:
Might be worth a comment that we expect this to be the case whenever we used `MK::Constant` instead of `MK::UnresolvedConstant`.